scripts/variantstore/wdl/extract/create_cohort_extract_data_table.py

```diff
@@ -285,55 +316,69 @@ def make_extract_table(fq_pet_vet_dataset,
-    TEMP_TABLE_TTL = f" OPTIONS( expiration_timestamp=TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL {temp_table_ttl_hours} HOUR)) "
+    TEMP_TABLE_TTL = f" OPTIONS( expiration_timestamp=TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL {TEMP_TABLE_TTL_HOURS} HOUR)) "
 
     print(f"Using {PET_VET_TABLE_COUNT} PET tables in {fq_pet_vet_dataset}...")
 
-    cohort = get_all_samples(fq_cohort_sample_names, fq_sample_mapping_table)
-    print(f"Discovered {len(cohort)} samples in {fq_cohort_sample_names}...")
+    # if we have a file of sample names, load it into a temporary table
+    if (sample_names_to_extract):
+        fq_sample_name_table = load_sample_names(sample_names_to_extract, fq_temp_table_dataset)
+    else:
+        fq_sample_name_table = fq_cohort_sample_names
+
+    # At this point one way or the other we have a table of sample names in BQ,
+    # join it to the sample_info table to drive the extract
+    create_extract_samples_table(fq_destination_table_samples, fq_sample_name_table, fq_sample_mapping_table)
 
-    make_new_vet_union_all(fq_pet_vet_dataset, fq_temp_table_dataset, cohort)
+    # pull the sample ids back down
+    sample_ids = get_all_sample_ids(fq_destination_table_samples)
+    print(f"Discovered {len(sample_ids)} samples in {fq_destination_table_samples}...")
+           
+    make_new_vet_union_all(fq_pet_vet_dataset, fq_temp_table_dataset, sample_ids)
 
     create_position_table(fq_temp_table_dataset, min_variant_samples)
-    make_new_pet_union_all(fq_pet_vet_dataset, fq_temp_table_dataset, cohort)
-    populate_final_extract_table(fq_temp_table_dataset, fq_destination_dataset, destination_table, fq_sample_mapping_table)
+    make_new_pet_union_all(fq_pet_vet_dataset, fq_temp_table_dataset, sample_ids)
+    populate_final_extract_table(fq_temp_table_dataset, fq_destination_table_data, fq_destination_table_samples)
   finally:
     dump_job_stats()
 
-  print(f"\nFinal cohort extract written to {fq_destination_dataset}.{destination_table}\n")
+  print(f"\nFinal cohort extract data written to {fq_destination_table_data}\n")
 
 if __name__ == '__main__':
   parser = argparse.ArgumentParser(allow_abbrev=False, description='Extract a cohort from BigQuery Variant Store ')
 
   parser.add_argument('--fq_petvet_dataset',type=str, help='project.dataset location of pet/vet data', required=True)
   parser.add_argument('--fq_temp_table_dataset',type=str, help='project.dataset location where results should be stored', required=True)
   parser.add_argument('--fq_destination_dataset',type=str, help='project.dataset location where results should be stored', required=True)
-  parser.add_argument('--destination_table',type=str, help='destination table', required=True)
-  parser.add_argument('--fq_cohort_sample_names',type=str, help='FQN of cohort table to extract, contains "sample_name" column', required=True)
+  parser.add_argument('--destination_cohort_table_prefix',type=str, help='prefix for destination tables', required=True)
```

Comment:
similar comment to above - passing around a prefix works well if PrepareCallset and ExtractCallset are only meant to be used with each other but from the perspective of someone that just wants to run PrepareCallset, they wouldn't know the fq names of the tables generated without reading the implementation and seeing that `__DATA` and `__SAMPLE` suffixes are being added.

In that regard, I think the explicit fq table names are nice but I think a comment in the argument's help section will also suffice. It could also be good to give an example of a valid prefix. It's not immediately clear if the prefix should contain just the table name prefix or the project and dataset as well.